share/spack/qa/setup-env-test.fish

```diff
@@ -90,9 +90,11 @@ function spt_succeeds
 
     set -l output (eval $argv 2>&1)
 
-    if test $status -ne 0
+    set cmd_status $status
```

Comment:
Is this still necessary? Your change on line 257 replacing the trap with the `function -p` should allow `$status` to work again. No?

Reply:
`$status` changes from command to command:

```#!/usr/bin/env fish

function example
  true
  echo "$status"
  false
  echo "$status"
end

example
```

echos `0` and `1`. So i'm just storing $status to print it later in `echo_red`